modules/gapi/include/opencv2/gapi/video.hpp

```diff
@@ -129,6 +129,128 @@ G_TYPED_KERNEL(GBackgroundSubtractor, <GMat(GMat, BackgroundSubtractorParams)>,
     }
 };
 
+/** @brief Structure for the Kalman filter's initialization parameters.*/
+
+struct GAPI_EXPORTS KalmanParams
+{
+    //! Type of the created matrices that should be CV_32F or CV_64F.
+    int type = CV_32F;
+    //! Dimensionality of the control vector.
+    int ctrlDim = 0;
+    //! Dimensionality of the state.
+    int dpDim = 2;
+    //! Dimensionality of the measurement.
+    int mpDim = 2;
+
+    // initial state
+
+    //! predicted state(x'(k)): x(k)=A*x(k-1)+B*u(k)
+    Mat statePre;
+    //! priori error estimate covariance matrix (P'(k)): P'(k)=A*P(k-1)*At + Q)*/
+    Mat errorCovPre;
+
+    // dynamic system description
+
+    //! state transition matrix (A)
+    Mat transitionMatrix;
+    //! measurement matrix (H)
+    Mat measurementMatrix;
+    //! process noise covariance matrix (Q)
+    Mat processNoiseCov;
+    //! measurement noise covariance matrix (R)
+    Mat measurementNoiseCov;
+    //! control matrix (B) (Optional: not used if there's no control)
+    Mat controlMatrix;
+
+    /** Full constructor
+    @param dp Dimensionality of the state.
+    @param mp Dimensionality of the measurement.
+    @param cp Dimensionality of the control vector. If it equals 0, dynamic system don't have external impact.
+    So controlMatrix should be empty.
+    @param tp Type of the created matrices that should be CV_32F or CV_64F.
+    */
+    KalmanParams(int dp, int mp, int cp = 0, int tp = CV_32F);
+};
+
+G_TYPED_KERNEL(GKalmanFilter, <GMat(GMat, GOpaque<bool>, GMat, KalmanParams)>, "org.opencv.video.KalmanFilter")
+{
+    static GMatDesc outMeta(const GMatDesc& measurement, const GOpaqueDesc&, const GMatDesc& control, const KalmanParams& kfParams)
+    {
+        GAPI_Assert(kfParams.type == CV_32F || kfParams.type == CV_64F);
+        GAPI_Assert(kfParams.dpDim > 0 && kfParams.mpDim > 0 && kfParams.ctrlDim >= 0);
+
+        GAPI_Assert(kfParams.statePre.type() == kfParams.type &&
+                    kfParams.errorCovPre.type() == kfParams.type &&
+                    kfParams.transitionMatrix.type() == kfParams.type &&
+                    kfParams.measurementMatrix.type() == kfParams.type &&
+                    kfParams.processNoiseCov.type() == kfParams.type &&
+                    kfParams.measurementNoiseCov.type() == kfParams.type);
+
+        if (!kfParams.controlMatrix.empty() && kfParams.ctrlDim > 0)
```

Comment:
I think we can put an assert here on controlMatrix is not empty since we have the overload with no control at all, it's better to use this overload if dynamic model doesn't involve control

Reply:
If so, should we impose restrictions for `kfParams.ctrlDim` too via asserts?
I mean, in this overload we can require `kfParams.ctrlDim > 0`, just as now it is required `kfParams.controlMatrix.empty() && kfParams.ctrlDim == 0` in the overload without control.
If you'll choose to apply this, please, make a note in documentation about it like `In this overload, it is restricted to set controlMatrix empty or ctrlDim field of KalmanParams equal 0. Use overload without control in case dynamic system don't have external impact.`

Reply:
Done.